src/main/java/com/mongodb/jdbc/MongoResultSet.java

```diff
@@ -159,7 +159,7 @@ public void close() throws SQLException {
         }
         cursor.close();
         closed = true;
-        if (statement != null && statement.isCloseOnCompletion()) {
+        if (statement != null && !statement.isClosed && statement.isCloseOnCompletion()) {
```

Comment:
Question about `(MongoStatement).isCloseOnCompletion()`: do we want to keep the `checkClosed()` call at the start of that method? I'm not sure there is a reason to check if the connection is closed or not to return the `closeOnCompletion` flag for a statement. Maybe instead of adding this check here, we should instead update the `isCloseOnCompletion` method?

Reply:
Oh, thanks for pointing that out! That's too bad 🤷‍♀️